From 8fae1bae7a6efa51d703753adeb0f240e71ea6e4 Mon Sep 17 00:00:00 2001
From: ryanking13 <def6488@gmail.com>
Date: Sun, 28 Jul 2024 18:15:17 +0900
Subject: [PATCH 1/1] Link openblas with modules that require f2c

Some fortran modules require symbols from f2c, which is provided by openblas.
This patch adds openblas as a dependency to the modules that require f2c symbols.

---
 scipy/integrate/meson.build | 2 +-
 scipy/optimize/meson.build  | 2 +-
 scipy/special/meson.build   | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/scipy/integrate/meson.build b/scipy/integrate/meson.build
index 4acc119a5..3c0e0f695 100644
--- a/scipy/integrate/meson.build
+++ b/scipy/integrate/meson.build
@@ -167,7 +167,7 @@ py3.extension_module('_dop',
   _dop_module,
   link_with: [dop_lib],
   c_args: [numpy_nodepr_api, Wno_unused_variable],
-  dependencies: [fortranobject_dep],
+  dependencies: [lapack, fortranobject_dep],
   link_args: version_link_args,
   install: true,
   link_language: 'fortran',
diff --git a/scipy/optimize/meson.build b/scipy/optimize/meson.build
index 964edad36..dd272801b 100644
--- a/scipy/optimize/meson.build
+++ b/scipy/optimize/meson.build
@@ -172,7 +172,7 @@ _slsqp = py3.extension_module('_slsqp',
   c_args: numpy_nodepr_api,
   fortran_args: fortran_ignore_warnings,
   link_args: version_link_args,
-  dependencies: [fortranobject_dep],
+  dependencies: [lapack, fortranobject_dep],
   install: true,
   link_language: 'fortran',
   subdir: 'scipy/optimize'
diff --git a/scipy/special/meson.build b/scipy/special/meson.build
index d166555f5..c0ca05f77 100644
--- a/scipy/special/meson.build
+++ b/scipy/special/meson.build
@@ -276,7 +276,7 @@ py3.extension_module('_specfun',
   specfun_module,
   c_args: numpy_nodepr_api,
   link_args: version_link_args,
-  dependencies: [fortranobject_dep],
+  dependencies: [lapack, fortranobject_dep],
   link_with: specfun_lib,
   link_language: 'fortran',
   install: true,
-- 
2.29.2.windows.2

